### PR TITLE
New: double clicking in an item in entity outliner does a camera go-to

### PR DESCRIPTION
**Pull request by LafLaurine**
## What does this PR do?

In most 3d authoring software, double clicking on an entity in the entity outliner window makes the camera focused on said entity in the viewport. This PR allows to reproduce this behavior.

Currently, when an entity is selected, you need to press Z to go to this entity.

This PR changes the default behavior of the double click :

- For an editor entity, for a prefab and for an entity inside a prefab : go to the selected entity

https://github.com/user-attachments/assets/c26a6cdf-e62b-4e6d-83f5-f4dafe1fecef

## How was this PR tested?

Local windows build, on the GnomebodysHome project. Selected an entity, a prefab and an entity inside a prefab.

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/EditorEntityUi/EditorEntityUiSystemComponent.h [hunk 1]
```diff
@@ -45,6 +46,7 @@ namespace AzToolsFramework
 
         private:
             EditorEntityUiHandlerId m_idCounter = 1;
+            AZStd::unique_ptr<EditorEntityUiHandlerBase> m_defaultEntityUiHandler;
```
Comment (LafLaurine) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/EditorEntityUi/EditorEntityUiSystemComponent.h:49]: Decided to rely on a default handler as the behavior is meant to be common to vast majority of cases. To register every entity not part of another handler in this "default" handler would introduce annoying logic and likely performance bottlenecks on huge levels . The default behavior can be opt-out when registering an handler, as done for the level root.

**Comment by LafLaurine** [comment]
> Didn't double clicking on a prefab previously focus that prefab so that your edits go to it? Is this a conflict?

Hello! Previously, double clicking on a prefab wasn't doing a camera go to, it is kinda an "enter edit mode" (all other entities become grayed out) so calling `Base` in the PrefabUiHandler is not conflicting with current behavior

**Review by lsemp3d — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/EditorEntityUi/EditorEntityUiHandlerBase.cpp [hunk 2]
```diff
@@ -120,6 +120,7 @@ namespace AzToolsFramework
 
     bool EditorEntityUiHandlerBase::OnOutlinerItemDoubleClick([[maybe_unused]] const QModelIndex& index) const
     {
+        AzToolsFramework::EditorRequestBus::Broadcast(&AzToolsFramework::EditorRequestBus::Events::GoToSelectedEntitiesInViewports);
```
Comment (AMZN-daimini) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/EditorEntityUi/EditorEntityUiHandlerBase.cpp:123]: The base Ui Handler is a stub class that is inherited from by other UI handlers, so as such it should not have behavior.

This code should be moved elsewhere. Ideally you would create another handler in-between the base and the prefab ui handler if you want this behavior to also be inherited by that.

**Review by AMZN-daimini — CHANGES_REQUESTED** [review]

**Comment by AMZN-daimini** [comment]
Also... the whole point of the Editor Entity Ui Handlers system is to provide alternate/additional behavior from the default for specific entities. So having a default handler is kind of pointless. You should instead just program the default behavior in the Outliner, right before it calls `OnOutlinerItemDoubleClick`.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/EditorEntityUi/EditorEntityUiHandlerBase.cpp (hunk 2 above)
Comment (LafLaurine) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/EditorEntityUi/EditorEntityUiHandlerBase.cpp]: Indeed I see your point! I modified the code based on your suggestion

Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/EditorEntityUi/EditorEntityUiHandlerBase.cpp]: Sounds better to me :) Hypothetically, we could also put it in the `else` branch of this if:
```
if (AZ::EntityId entityId = GetEntityIdFromIndex(index); auto entityUiHandler = m_editorEntityUiInterface->GetHandler(entityId))
```
this way handlers overriding that function would not get that behavior.
I think it makes sense to always apply it for now though, and we can make the change if any use-case arises where that is not what we want (but we would have to call that function manually in the Prefab handler then).

TLDR, thanks for changing this, looks good :)
